src/query/functions/src/scalars/comparison.rs

```diff
@@ -445,48 +444,13 @@ fn register_like(registry: &mut FunctionRegistry) {
     registry.register_passthrough_nullable_2_arg::<VariantType, StringType, BooleanType, _, _>(
         "like",
         |_, _, _| FunctionDomain::Full,
-        variant_vectorize_like(|val, pat, _, pattern_type| {
-            match &pattern_type {
-                LikePattern::OrdinalStr => {
-                    if let Some(s) = jsonb::as_str(val) {
-                        LikePattern::ordinal_str(s.as_bytes(), pat)
-                    } else {
-                        false
-                    }
-                }
-                LikePattern::EndOfPercent => {
-                    // fast path, can use starts_with
-                    if let Some(s) = jsonb::as_str(val) {
-                        LikePattern::end_of_percent(s.as_bytes(), pat)
-                    } else {
-                        false
-                    }
-                }
-                LikePattern::StartOfPercent => {
-                    // fast path, can use ends_with
-                    if let Some(s) = jsonb::as_str(val) {
-                        LikePattern::start_of_percent(s.as_bytes(), pat)
-                    } else {
-                        false
-                    }
-                }
-                LikePattern::SurroundByPercent => {
-                    jsonb::traverse_check_string(val, |v| LikePattern::surround_by_percent(v, pat))
-                }
-                LikePattern::SimplePattern(simple_pattern) => {
-                    jsonb::traverse_check_string(val, |v| {
-                        LikePattern::simple_pattern(
-                            v,
-                            simple_pattern.0,
-                            simple_pattern.1,
-                            &simple_pattern.2,
-                        )
-                    })
-                }
-                LikePattern::ComplexPattern => {
-                    jsonb::traverse_check_string(val, |v| LikePattern::complex_pattern(v, pat))
-                }
-            }
+        variant_vectorize_like(|val, pattern_type| match pattern_type {
+            LikePattern::OrdinalStr(_)
+            | LikePattern::StartOfPercent(_)
+            | LikePattern::EndOfPercent(_)
+            | LikePattern::Constant(_) => pattern_type.compare(val),
```

Comment:
`val` is binary format, we need use `jsonb::as_str` function to convert it to a string value, and then run `compare` function.